File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int ans = -1, n = s.size();
        unordered_map<char, vector<int>> idxLists;
        
        for (int i = 0; i < n; ++i){
            idxLists[s[i]].push_back(i);
        }
        
        for (auto& [_, indices]: idxLists){
            ans = max(ans, indices.back() - indices[0] - 1);
        }
        
        return ans;
    }
};

File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        first_index, ans = dict(), -1
        
        for i in range(len(s)):
            if s[i] in first_index:
                ans = max(ans, i - first_index[s[i]] - 1)
            else:
                first_index[s[i]] = i
        
        return ans